.github/workflows/build-pr.yml

```diff
@@ -4,19 +4,67 @@ on:
   push:
     branches: [ "master" ]
   pull_request:
-    branches: [ "master" ]
+    branches: 
+      - master
+      - 'feature/**'
 
 jobs:
-  build:
-
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read      # required by actions/checkout
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install poetry
```

Comment:
Isn't it recommended to use `pipx` to segregate environments?

Reply:
Sure, here as each jb is ran on a different machine I don't think it would make a difference but it is a good practice.